internal/security/config/command/proxy/deluser/command.go

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package deluser
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/security/config/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/security/proxy/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/secretstoreclient"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+const (
+	CommandName string = "deluser"
+)
+
+type cmd struct {
+	loggingClient logger.LoggingClient
+	client        internal.HttpCaller
+	configuration *config.ConfigurationStruct
+	tokenType     string
+	username      string
+}
+
+func NewCommand(
+	lc logger.LoggingClient,
+	configuration *config.ConfigurationStruct,
+	args []string) (interfaces.Command, error) {
+
+	cmd := cmd{
+		loggingClient: lc,
+		client:        secretstoreclient.NewRequestor(lc).Insecure(),
+		configuration: configuration,
+	}
+	var dummy string
+
+	flagSet := flag.NewFlagSet(CommandName, flag.ContinueOnError)
+	flagSet.StringVar(&dummy, "confdir", "", "") // handled by bootstrap; duplicated here to prevent arg parsing errors
+
+	flagSet.StringVar(&cmd.tokenType, "token-type", "", "Type of token to create: jwt or oauth2")
+	flagSet.StringVar(&cmd.username, "user", "", "Username of the user to delete")
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+	if cmd.tokenType == "" {
+		return nil, fmt.Errorf("%s proxy deluser: argument --token-type is required", os.Args[0])
+	}
+	if cmd.tokenType != interfaces.JwtTokenType && cmd.tokenType != interfaces.OAuth2TokenType {
+		return nil, fmt.Errorf("%s proxy deluser: argument --token-type must be either 'jwt' or 'oauth2'", os.Args[0])
+	}
+	if cmd.username == "" {
+		return nil, fmt.Errorf("%s proxy deluser: argument --user is required", os.Args[0])
+	}
+
+	return &cmd, err
+}
+
+func (c *cmd) Execute() (int, error) {
+	// Delete Kong consumer
+	// https://docs.konghq.com/2.1.x/admin-api/#delete-consumer
```

Comment:
We never updated to Kong 2.1.x, is the API backwards compatible with 2.0.x?

Reply:
Seems to still work.